app/Http/Middleware/Checks/IsInstalled.php

```diff
@@ -19,6 +21,8 @@ public function assert(): bool
 			return
 				!file_exists(base_path('.NO_SECURE_KEY')) &&
 				Schema::hasTable('configs');
+		} catch (QueryException $e) {
+			return !Str::contains($e->getMessage(), 'SQLSTATE[HY000] [1045]');
```

Comment:
I can somehow understand that the method `assert` should not throw an exception, but always return either `true` or `false`. So I would have expected something like

```php
try {
    // do whatever needs to be done and return true or false
} catch (BindingResolutionException $e)  {
    // re-throw as this should never happen anyway, because this indicates a programming error
} catch {
    // return false for any other not rethrown message because we could obviously _not_ assert what we wanted to assert
    return false;
}
```

In particular, I do not understand why why the return value depends on whether the message contains `''SQLSTATE[HY000] [1045]"` or not. After googling, I found out that this message is thrown by MySQL if DB authentication failed.

First of all, this should be part of a comment above this line.

Secondly, there should be an explanation _why_ we return `true` if there has been any other SQL exception than an authentication problem.

Reply:
I don't see what kind of other SQL exception we could have aside from this one.
I do not want to make the catch too broad in order to avoid useless redirection to Install.